common/common.go

```diff
@@ -123,6 +123,19 @@ const (
 	AnnotationValueManagedByArgoCD = "argocd.argoproj.io"
 	// ResourcesFinalizerName the finalizer value which we inject to finalize deletion of an application
 	ResourcesFinalizerName = "resources-finalizer.argocd.argoproj.io"
+
+	// AnnotationKeyRefreshPathUpdatesOnly is an annotation used by the api-server when processing webhooks
+	// it tells the api server to only trigger app refreshes when the webhook contains changes for
+	// the files under the path defined at `.Spec.Source.Path` in the application resource.
+	// This is enough for most apps, but some Kustomize installations may need to trigger
+	// on paths outside of the app path. In these cases, use AnnotationKeyRefreshPrefix to specify the path.
+	// The associated value should be a string of "true"
+	AnnotationKeyRefreshPathUpdatesOnly = "argocd.argoproj.io/refresh-on-path-updates-only"
+	// AnnotationKeyRefreshPrefix is an annotation used by the api-server when processing webhooks
+	// it tells the api server to only trigger app refreshes when the webhook contains changes for
+	// files under the given path.
+	// The associated value should be a file path.
+	AnnotationKeyRefreshPrefix = "argocd.argoproj.io/refresh-prefix"
```

Comment:
Couple suggestions here. 
* Rename annotation to make more explicit that it is related to manifest generation. E.g. `argocd.argoproj.io/manifest-generation-path`.


* Can we treat value in this annotation as a relative path and support `.` value? E.g. 

```yaml
annotations:
  argocd.argoproj.io/refresh-prefix: guestbook
```

In this case, user can add `argocd.argoproj.io/refresh-prefix: .` and we probably don't need `argocd.argoproj.io/refresh-on-path-updates-only` at all.